sdk/storage/azure-storage-blobs/src/blob_responses.cpp

```diff
@@ -58,4 +59,48 @@ namespace Azure { namespace Storage { namespace Blobs {
     };
   }
 
+  void ListBlobsPageResult::OnNextPage(const Azure::Core::Context& context)
+  {
+    auto contextRef = std::cref(context);
+    if (&context == &Azure::Core::Context::GetApplicationContext())
+    {
+      // Called via iterator increment, use the same context as the last call.
```

Comment:
Care to elaborate Jeff why storing a context field not a good idea and why comparing the ref to the root context is dangerous?

I haven't seen this coding pattern with Context  elsewhere so it stood out to me as well.

Reply:
A context field means that the PageResponse is holding on to a Context so that it can be reused. But, we don't want to reuse them: each I/O operation gets its very own Context tree and this is important because the customer might want to specify that the first page operation should exist at 100pm and then the next page operation should expire at 1:10pm, etc.

If the first Context is used for all operations, then ALL operations must complete by 1pm and this really bad because the customer probably does not know how long processing each page's items will take and this time will be included in the Context's deadline and this is not the semantic meaning of Contexts. A Context is passed to a SINGLE I/O operation to control this operation's deadline.

The reason why comparing to root is dangerous/bad is because not all Contexts have the same root. So, this "magic code" has baked in an assumption that if the application context root is canceled then this operation should be canceled but the customer has never specified this and the behavior is unpredictable to them.